### PR TITLE
feat(HACBS-2405): add tests for push-sbom-to-pyxis

### PR DESCRIPTION
**Pull request by mmalina**
These tests need some mocking.

Mocks are needed when we want to test tasks which call external services. In this case we want to mock calls of cosign and upload_sbom.

To achieve that, a new file
with mock function definitions is added to the tests/ directory. The mocks are then injected at the beginning of the task's scripts.

This is an alternative solution to https://github.com/redhat-appstudio/release-service-bundles/pull/159

### CODE REVIEW

File: CONTRIBUTING.md
Comment (johnbieren) [inline CONTRIBUTING.md:1]: Can you make this its own PR?

Comment (mmalina) [reply CONTRIBUTING.md:1]: ok, here's a separate PR for the docs: https://github.com/redhat-appstudio/release-service-bundles/pull/187

**Review by johnbieren — APPROVED** [review]
